.github/workflows/ci.yaml

```diff
@@ -6,22 +6,17 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - uses: actions/cache@v1
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8.4.0
+      - uses: actions/setup-node@v3
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          cache: "pnpm"
 
       - name: Install project dependencies and build
         run: |
-          yarn --ignore-engines --frozen-lockfile
-          yarn build
+          pnpm install --frozen-lockfile
+          npm run build
```

Comment:
You can use [`pnpm run`](https://pnpm.io/cli/run)

Reply:
As same as `npm run` below.

Reply:
Here I choose NPM instead of pnpm because almost all the NodeJS users have installed npm, so we'd use npm wherever we could.

Which do you think is more appropriate, "only one package manager" or "use npm wherever we can"?

@zhangyouxin @IronLu233

Reply:
If the user want to run our project, he/she must install `pnpm` beforehand. I vote for “only one package manager”.